src/views/pages/App.js

```diff
@@ -18,13 +23,26 @@ class App extends React.Component {
     );
   }
 
-  setUpperText = e => {};
+  setUpperText = e => {
+    this.setState({upperText: e.target.value});
+  };
 
-  setLowerText = e => {};
+  setLowerText = e => {
+    this.setState({lowerText: e.target.value});
+  };
 
-  onSelect = e => {};
+  handleClick = (image, e) => {
+    this.setState({selectedImage: image});
+  };
 
-  onClick = e => {};
+  handleSubmit = e => {
+    e.preventDefault();
```

Comment:
Why do we need this?

Reply:
Submit buttons redirect you by default, and we don't want that in this situation. To prevent this we must call the preventDefault method of the event.